core/src/test/java/org/elasticsearch/index/query/GeoBoundingBoxQueryBuilderTest.java

```diff
@@ -0,0 +1,222 @@
+        return randomFrom(values);
+    }
+
+    @Override
+    protected Query doCreateExpectedQuery(GeoBoundingBoxQueryBuilder qb, QueryParseContext parseContext) throws IOException {
+        if (parseContext.mapperService().smartNameFieldType("mapped_geo", parseContext.getTypes()) == null){
+            return new MatchNoDocsQuery();
+        }
+
+        MappedFieldType fieldType = parseContext.fieldMapper(qb.fieldName());
+        GeoPointFieldMapper.GeoPointFieldType geoFieldType = ((GeoPointFieldMapper.GeoPointFieldType) fieldType);
+
+        Query result;
+        if (Type.INDEXED.equals(qb.type())) {
+            result = IndexedGeoBoundingBoxQuery.create(qb.topLeft(), qb.bottomRight(), geoFieldType);
+        } else {
+            throw(new UnsupportedOperationException("Cannot yet test MEMORY GeoBoundingBox queries."));
+        }
+
+        return result;
+    }
+
+    @Test
+    public void testValidation() {
+        GeoBoundingBoxQueryBuilder builder = new GeoBoundingBoxQueryBuilder("");
+        builder.topLeft(0.0, 0.0);
+        builder.bottomRight(0.0, 0.0);
+        assertNotNull("Empty string as fieldname should not be valid.", builder.validate());
+
+        builder = new GeoBoundingBoxQueryBuilder(null);
+        builder.topLeft(0.0, 0.0);
+        builder.bottomRight(0.0, 0.0);
+        assertNotNull("null as fieldname should not be valid.", builder.validate());
+
+        // add remainder of interval, make interval exclusive
+        double invalidTop = randomFrom(invalidRandomDouble(MIN_LAT, MAX_LAT));
+        double invalidBottom = randomFrom(invalidRandomDouble(MIN_LAT, MAX_LAT));
+        double invalidLeft = randomFrom(invalidRandomDouble(MIN_LON, MAX_LON));
+        double invalidRight = randomFrom(invalidRandomDouble(MIN_LON, MAX_LON));
+
+        builder = new GeoBoundingBoxQueryBuilder("fieldname");
+        builder.topLeft(invalidTop, 0.0);
+        builder.bottomRight(0.0, 0.0);
+        assertNotNull("Validation should detect invalid top coordinate: " + invalidTop, builder.validate());
+
+        builder = new GeoBoundingBoxQueryBuilder("fieldname");
+        builder.topLeft(0.0, invalidLeft);
+        builder.bottomRight(0.0, 0.0);
+        assertNotNull("Validation should detect invalid left coordinate: " + invalidLeft, builder.validate());
+
+        builder = new GeoBoundingBoxQueryBuilder("fieldname");
+        builder.topLeft(0.0, 0.0);
+        builder.bottomRight(invalidBottom, 0.0);
+        assertNotNull("Validation should detect invalid bottom coordinate: " + invalidBottom, builder.validate());
+
+        builder = new GeoBoundingBoxQueryBuilder("fieldname");
+        builder.topLeft(0.0, 0.0);
+        builder.bottomRight(0.0, invalidRight);
+        assertNotNull("Validation should detect invalid right coordinate: " + invalidRight, builder.validate());
+
+        double bottom = randomDoubleBetween(MIN_LAT, MAX_LAT);
+        while (bottom == MAX_LAT) {
```

Comment:
I don't quiet get what this while loop is doing here? If bottom should not be MAX_LAT, why not subtract small delta from it when generating it? Am I missing something?